converters/toZigbee.js

```diff
@@ -190,6 +190,294 @@ const converters = {
+                    zclData: {
+                        dataType: 0x29, // dataType
+                        attrData: Math.round(value) * 100,
+                        numoftrans: value.numoftrans,
+                        dayofweek: value.dayofweek,
+                        mode: value.mode,
+                        thermoseqmode: value.thermoseqmode,
+                    },
+                    cfg: cfg.default,
+                };
+            } else if (type === 'get') { // MAC transaction expired.
+                return {
+                    cid: cid,
+                    cmd: 'getWeeklySchedule',
+                    cmdType: 'functional',
+                    zclData: [
+                        {attrId: zclId.attr(cid, attrId).value},
+                        // {daystoreturn: value.daystoreturn},
+                        // {modetoreturn: value.modetoreturn},
+                    ],
+                    cfg: cfg.default,
+                };
+            }
+        },
+    },
+    thermostat_clearWeeklySchedule: { // not tested
+        key: 'clearWeeklySchedule',
+        attr: [],
+        convert: (key, value, message, type) => {
+            return {
+                cid: 'hvacThermostat',
+                cmd: 'clearWeeklySchedule',
+                type: 'functional',
+                zclData: {},
+            };
+        },
+    },
+    thermostat_getRelayStatusLog: { // not tested
+        key: 'getRelayStatusLog',
+        attr: [],
+        convert: (key, value, message, type) => {
+            return {
+                cid: 'hvacThermostat',
+                cmd: 'getRelayStatusLog',
+                type: 'functional',
+                zclData: {},
+            };
+        },
+    },
+    thermostat_getWeeklyScheduleRsp: { // not tested
+        key: 'getWeeklyScheduleRsp',
+        attr: [],
+        convert: (key, value, message, type) => {
+            return {
+                cid: 'hvacThermostat',
+                cmd: 'getWeeklyScheduleRsp',
+                type: 'functional',
+                zclData: {
+                    numoftrans: value.numoftrans,
+                    dayofweek: value.dayofweek,
+                    mode: value.mode,
+                    thermoseqmode: value.thermoseqmode,
+                },
+            };
+        },
+    },
+    thermostat_getRelayStatusLogRsp: { // not tested
+        key: 'getRelayStatusLogRsp',
+        attr: [],
+        convert: (key, value, message, type) => {
+            return {
+                cid: 'hvacThermostat',
+                cmd: 'getRelayStatusLogRsp',
+                type: 'functional',
+                zclData: {
+                    timeofday: value.timeofday,
+                    relaystatus: value.relaystatus,
+                    localtemp: value.localtemp,
+                    humidity: value.humidity,
+                    setpoint: value.setpoint,
+                    unreadentries: value.unreadentries,
```

Comment:
Please change to: time_of_day, relay_status, local_temperature, unread_entries (zigbee2mqtt uses snake casing for MQTT payloads)